web/packages/shared/components/AccessRequests/NewRequest/RequestCheckout/RequestCheckout.tsx

```diff
@@ -193,9 +208,19 @@ export function RequestCheckout<T extends PendingListItem>({
     data.length === 0 ||
     createAttempt.status === 'processing' ||
     isInvalidRoleSelection ||
-    fetchResourceRequestRolesAttempt.status === 'failed' ||
+    (fetchResourceRequestRolesAttempt.status === 'failed' &&
+      hasUnsupportedKubeRequestModes) ||
+    requiresNamespaceSelect ||
+    fetchResourceRequestRolesAttempt.status === 'processing';
+
+  const cancelBtnDisabled =
+    createAttempt.status === 'processing' ||
     fetchResourceRequestRolesAttempt.status === 'processing';
 
+  const numResourcesSelected = data.filter(item =>
```

Comment:
`num` makes it seem like the resulting value is a number, but it's an array. Finding a better name for `data` and `filteredData` would help with naming here too, see my comment from the other PR. https://github.com/gravitational/teleport/pull/47347#discussion_r1806528349